modules/adapters/slurm-collector/src/main.rs

```diff
@@ -0,0 +1,382 @@
+use serde::{Deserialize, Serialize};
+use std::fs::File as FsFile;
+use std::io::{BufRead, BufReader};
+use std::process::{Command, Stdio};
+use thiserror::Error;
+use tokio::time::{self, Duration};
+use tonic::transport::Channel;
+
+#[derive(Error, Debug)]
+enum AppError {
+    #[error("Sacct command failed: {0}")]
+    SacctError(String),
+
+    #[error("Failed to parse sacct output: {0}")]
+    InvalidSacctOutput(String),
+
+    #[error("Date parsing error: {0}")]
+    DateParseError(#[from] chrono::ParseError),
+
+    #[error("IO error: {0}")]
+    IoError(#[from] std::io::Error),
+
+    #[error("Serialization error: {0}")]
+    SerializationError(#[from] serde_json::Error),
+
+    #[error("Error setting up grpc connection : {0}")]
+    TonicConnectionError(#[from] tonic::transport::Error),
+
+    #[error("Tonic error: {0}")]
+    TonicStatusError(#[from] tonic::Status),
+
+    #[error("Ingestion was rejected by the server for some records")]
+    IngestError,
+}
+
+type Result<T> = std::result::Result<T, AppError>;
+
+#[derive(clap::Parser, Debug)]
+#[clap(author, version, about, long_about = None)]
+struct AppConfig {
+    #[clap(long, default_value = "300", help = "Polling interval in seconds")]
+    poll_interval: u64,
+
+    #[clap(long, env = "METEROID_INGEST_ENDPOINT", help = "API endpoint URL")]
+    api_endpoint: String,
+
+    #[clap(
+        long,
+        env = "METEROID_API_KEY",
+        hide_env_values = true,
+        help = "API key for authentication"
+    )]
+    api_key: String,
+
+    #[clap(long, default_value = "checkpoint.db", help = "Path to the state file")]
+    state_file: String,
+
+    #[clap(
+        long,
+        default_value = "200",
+        help = "Number of records to process in each batch"
+    )]
+    batch_size: usize,
+
+    #[clap(
+        long,
+        default_value = "2020-01-01T00:00:00Z",
+        help = "Initial checkpoint date in RFC 3339 format"
+    )]
+    initial_checkpoint: String,
+}
+
+type GrpcClient = EventsServiceClient<LayeredApiClientService>;
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+struct SacctData {
+    id: String,
+    job_id: String,
+    account: String,
+    start_time: DateTime<Utc>,
+    elapsed_seconds: i64,
+    state: String,
+    end_time: DateTime<Utc>,
+    req_cpu: i64,
+    req_mem: i64,
+    partition: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Checkpoint {
+    last_processed_time: DateTime<Utc>,
+    processed_jobs: Vec<String>,
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    env_logger::Builder::from_env(env_logger::Env::default().default_filter_or("info")).init();
+    info!("Starting Slurm sacct processor");
+
+    let config = AppConfig::parse();
+
+    let mut client = build_client(&config).await?;
+    let mut interval = time::interval(Duration::from_secs(config.poll_interval));
+
+    let sacct_executor = SacctExecutorImpl;
+    loop {
+        interval.tick().await;
```

Comment:
why not after the processing? are we ok with waiting 5 minutes on the startup?
prolly for production it doesn't matter much but when testing things might be helpful to process on startup.

Reply:
ok, the first tick is not gonna be instant so not an issue here.